addons/binding/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/internal/ExecHandlerFactory.java

```diff
@@ -21,14 +21,17 @@
 import org.eclipse.smarthome.core.thing.ThingTypeUID;
 import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
 import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
 import org.openhab.binding.exec.internal.handler.ExecHandler;
+import org.osgi.service.component.annotations.Component;
 
 /**
  * The {@link ExecHandlerFactory} is responsible for creating things and thing
  * handlers.
  *
  * @author Karel Goderis - Initial contribution
  */
+@Component(service = ThingHandlerFactory.class, configurationPid = "binding.exec")
```

Comment:
Can you also add `@NonNullByDefault`. It should be easy to add as well. :-)

Reply:
Of course.